File: Onboarding.js
/**
 * The code below will read all the game information for you.
 * On each game turn, information will be available on the standard input, you will be sent:
 * -> the total number of visible enemies
 * -> for each enemy, its name and distance from you
 * The system will wait for you to write an enemy name on the standard output.
 * Once you have designated a target:
 * -> the cannon will shoot
 * -> the enemies will move
 * -> new info will be available for you to read on the standard input.
 **/


// game loop
while (true) {
    var count = parseInt(readline()); // The number of current enemy ships within range
    var big_threat;
    var big_dist = 10000;
    for (var i = 0; i < count; i++) {
        var inputs = readline().split(' ');
        var enemy = inputs[0]; // The name of this enemy
        var dist = parseInt(inputs[1]); // The distance to your cannon of this enemy
        if(dist<big_dist) {
            big_dist = dist;
            big_threat = enemy;
        }
    }

    // Write an action using print()
    // To debug: printErr('Debug messages...');

    print(big_threat); // The name of the most threatening enemy (HotDroid is just one example)

}

File: onboarding.ml
(* CodinGame planet is being attacked by slimy insectoid aliens. *)
(* <--- *)
(* Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player. *)

let comparator (dist1, enemy1) (dist2, enemy2) = 
    if dist1 > dist2 then enemy2 else enemy1;;

(* game loop *)
while true do
    let enemy1 = input_line stdin in (* name of enemy 1 *)
    let dist1 = int_of_string (input_line stdin) in (* distance to enemy 1 *)
    let enemy2 = input_line stdin in (* name of enemy 2 *)
    let dist2 = int_of_string (input_line stdin) in (* distance to enemy 2 *)
    
    (* Write an action using print_endline *)
    (* To debug: prerr_endline "Debug message"; *)
    
    print_endline (comparator (dist1, enemy1) (dist2, enemy2)); (* You have to output a correct ship name to shoot ("Buzz", enemy1, enemy2, ...) *)
    ();
done;
